### DISCUSSION TITLE
Invalid redirect_uri when hosting SPA on Netlify

### DISCUSSION THREAD

**Discussion by benjamin-lindsey** [iTwin/itwinjs-core#2898] [Q&A] [answered]
Hi,
I'm working on a custom iTwin viewer. It works great on localhost. But when it's hosted on Netlify, I'm redirected to a 400 error page when signing in.
![error_redirect_uri](https://user-images.githubusercontent.com/49169565/145685186-06ef97e0-5f9b-473f-b297-4c3c4a840ed3.png)

 I followed the [medium](https://medium.com/itwinjs/deploying-the-itwin-viewer-to-a-web-host-d45c5cfdf0cf) article. All of the environment variables are identical in the .env file, Netlify's environment variable UI, and the app in the "My Apps" page. The netlify.toml is identical to the one in the medium article. 

I've set the redirect URI as my website url with `/signin-callback` appended, just like it's done with localhost. I've triple-checked all of the variables. What could be going wrong and is there any way to get a more descriptive error message?

Thank you for your help!

**Comment by kckst8** [comment]
Hi @benjamin-lindsey, do you have the redirects added to the client that you are using for your website as well? https://developer.bentley.com/my-apps/

![image](https://user-images.githubusercontent.com/6283674/145686675-0e8f10b4-146c-4fea-a578-a69b6d3e12c5.png)

**Reply by benjamin-lindsey** [reply]
Hi @kckst8, yes, I do. It was also set in the .env and Netlify environment variable IMJS_AUTH_CLIENT_REDIRECT_URI.

**Comment by benjamin-lindsey** [comment, marked as answer]
@kckst8 - Wow, I triple-checked everything _but_ the client ID. I just assumed it was the correct one since localhost worked and I had been using it for a while. Sheesh, well... thank you for your time!
